core/src/main/java/org/infinispan/interceptors/CacheLoaderInterceptor.java

```diff
@@ -93,46 +87,22 @@ protected void injectDependencies(PersistenceManager clm, EntryFactory entryFact
 
    @Override
    public Object visitApplyDeltaCommand(InvocationContext ctx, ApplyDeltaCommand command) throws Throwable {
-      if (enabled) {
-         Object key;
-         if ((key = command.getKey()) != null) {
-            loadIfNeeded(ctx, key, command);
-         }
-      }
-      return invokeNextInterceptor(ctx, command);
+      return visitDataCommand(ctx, command);
    }
 
    @Override
    public Object visitPutKeyValueCommand(InvocationContext ctx, PutKeyValueCommand command) throws Throwable {
-      if (enabled) {
-         Object key;
-         if ((key = command.getKey()) != null) {
-            loadIfNeeded(ctx, key, command);
```

Comment:
I agree as well (was writing up something else but this is short and sweet)

Also the InvalidateL1Command shouldn't exist in the store since we don't put it there on initial insert and they aren't passivated as of https://github.com/infinispan/infinispan/pull/2824